app/dashboard/views.py

```
@@ -1136,31 +1132,31 @@ def profile(request, handle):
             context['activities'] = paginator.get_page(page)
 
             return TemplateResponse(request, 'profiles/profile_activities.html', context, status=status)
-        else:
-            context = profile.to_dict(tips=False)
-            all_activities = context.get('activities')
-            activity_tabs = []
 
-            for tab, name in tabs:
-                activities = profile_filter_activities(all_activities, tab)
-                activities_count = activities.count()
+        context = profile.to_dict(tips=False)
+        all_activities = context.get('activities')
+        activity_tabs = []
+
+        for tab, name in tabs:
+            activities = profile_filter_activities(all_activities, tab)
+            activities_count = activities.count()
 
-                if activities_count == 0:
-                    continue
+            if activities_count == 0:
+                continue
 
-                paginator = Paginator(activities, 10)
+            paginator = Paginator(activities, 10)
 
-                obj = {}
-                obj['id'] = tab
-                obj['name'] = name
-                obj['activities'] = paginator.get_page(1)
-                obj['count'] = activities_count
+            obj = {}
+            obj['id'] = tab
+            obj['name'] = name
+            obj['activities'] = paginator.get_page(1)
+            obj['count'] = activities_count
 
-                activity_tabs.append(obj)
+            activity_tabs.append(obj)
 
             context['activity_tabs'] = activity_tabs
 
-    except (Http404, ProfileHiddenException):
+    except (Http404, ProfileHiddenException, ProfileNotFoundException):
```

Catch all exceptions raised by `profile_helper`